### PR TITLE
39 create session screen

### PR DESCRIPTION
**Pull request by DimaK21**
Экран создания сессии
добавлена библиотека com.google.android.flexbox для FlexboxLayoutManager (элементы в строку, при достижении конца экрана - новая строка)

### CODE REVIEW

File: app/src/main/java/com/davay/android/feature/createsession/presentation/createsession/CreateSessionFragment.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                }
+            }
+        })
+        binding.toolBar.setStartIconClickListener {
+            findNavController().navigateUp()
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
```
Comment (Eka666Sed) [inline app/src/main/java/com/davay/android/feature/createsession/presentation/createsession/CreateSessionFragment.kt:80]: Привет! Чтобы правильно освобождать ресурсы в onDestroyView, нужно вызывать super.onDestroyView() после отключения tabMediator:
override fun onDestroyView() {
        tabMediator?.detach()
        tabMediator = null
        super.onDestroyView()
        }

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                            com.davai.uikit.R.drawable.divider,
+                            requireContext().theme
+                        )!!
+                    )
+                }
+        )
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
```
Comment (Eka666Sed) [inline app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt:51]: При уничтожении onDestroyView я бы добавила binding.rvCompilations.adapter = null на всякий случай.

**Review by Eka666Sed — APPROVED** [review]

**Comment by Eka666Sed** [comment]
Screenshot_2024-05-19-17-06-42-39_2bfdc68e41af29879a10df8e723fd781
Тут под вопросом: меня продолжает смущать, как ведут себя Toolbar и Navbar. Потом надо на Activity проверить;)

**Comment by DimaK21** [comment]
> Screenshot_2024-05-19-17-06-42-39_2bfdc68e41af29879a10df8e723fd781
> Тут под вопросом: меня продолжает смущать, как ведут себя Toolbar и Navbar. Потом надо на Activity проверить;)

Решили вроде делать системные отступы сверху и снизу.
Я добавил их в PR с экраном регистрации. То есть экран чуть сожмется по вертикали, если тот PR будет принят

File: app/src/main/java/com/davay/android/feature/createsession/presentation/createsession/CreateSessionFragment.kt (hunk 1 above)
Comment (DimaK21) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/createsession/CreateSessionFragment.kt]: поправил

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt (hunk 2 above)
Comment (DimaK21) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt:52]: добавил

File: uikit/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -20,6 +20,7 @@
     <dimen name="margin_24">24dp</dimen>
     <dimen name="margin_20">20dp</dimen>
     <dimen name="margin_32">32dp</dimen>
+    <dimen name="margin_16m">-16dp</dimen>
```
Comment (GoetzDeBouville) [inline uikit/src/main/res/values/dimens.xml:23]: я уже стал называть такие размеры с приставкой negative:
margin_negative_16

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import com.davay.android.databinding.FragmentCompilationsBinding
+import com.davay.android.di.ScreenComponent
+import com.davay.android.feature.createsession.di.DaggerCreateSessionFragmentComponent
+import com.davay.android.feature.createsession.domain.model.Compilation
+
+class CompilationsFragment : BaseFragment<FragmentCompilationsBinding, CompilationsViewModel>(
+    FragmentCompilationsBinding::inflate
+) {
+    override val viewModel: CompilationsViewModel by injectViewModel<CompilationsViewModel>()
+    private var compilationAdapter: CompilationsAdapter? = null
```
Comment (GoetzDeBouville) [inline app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt:18]: предлагаю обсудить как поступать с созданием/обнулением адаптера, потому что мне кажется что без сильных референсов типа контекст он будет уничтожен вместе с фрагментом

**Review by GoetzDeBouville — APPROVED** [review]

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt (hunk 2 above)
Comment (GoetzDeBouville) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt:64]: кажется, что он будет уничтожен, так как super.onDestroyView() уже уничтожает представление, разве нет?

File: uikit/src/main/res/layout/toolbar_view.xml [hunk 5]
```diff
@@ -3,29 +3,38 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
-    android:layout_height="@dimen/toolbar_height_64"
+    android:layout_height="wrap_content"
```
[the remark this answers is not in the document]
Comment (DimaK21) [reply uikit/src/main/res/layout/toolbar_view.xml:6]: 56 влита уже, сегодня перепроверил, ничего нового не добавляется

File: app/src/main/java/com/davay/android/app/MainActivity.kt [hunk 6]
```diff
@@ -39,4 +58,45 @@ class MainActivity : AppCompatActivity() {
...
+            val insets = windowInsets.getInsets(WindowInsetsCompat.Type.statusBars())
+            v.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+                topMargin = insets.top
+            }
+            WindowInsetsCompat.CONSUMED
+        }
+    }
+
+    fun updateBanner(text: String, type: Int) {
+        binding.banner.setBannerText(text)
```
[the remark this answers is not in the document]
Comment (DimaK21) [reply app/src/main/java/com/davay/android/app/MainActivity.kt]: сделал

File: app/src/main/java/com/davay/android/app/MainActivity.kt [hunk 7]
```diff
@@ -39,4 +58,45 @@ class MainActivity : AppCompatActivity() {
...
+        val mediumAnimationDuration = resources.getInteger(android.R.integer.config_mediumAnimTime)
+        binding.banner.apply {
+            alpha = 0f
+            visibility = View.VISIBLE
+            animate()
+                .alpha(1f)
+                .setDuration(mediumAnimationDuration.toLong())
+                .setListener(null)
+        }
+        binding.banner.postDelayed(BANNER_TIME_MS) {
```
[the remark this answers is not in the document]
Comment (DimaK21) [reply app/src/main/java/com/davay/android/app/MainActivity.kt]: сделал

File: app/src/main/java/com/davay/android/app/MainActivity.kt [hunk 8]
```diff
@@ -39,4 +58,45 @@ class MainActivity : AppCompatActivity() {
...
+                .setListener(object : AnimatorListenerAdapter() {
+                    override fun onAnimationEnd(animation: Animator) {
+                        binding.banner.visibility = View.GONE
+                    }
+                })
+        }
+    }
+
+    companion object {
+        const val BANNER_TIME_MS = 2000L
```
[the remark this answers is not in the document]
Comment (DimaK21) [reply app/src/main/java/com/davay/android/app/MainActivity.kt]: сделал

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,35 @@
+package com.davay.android.feature.createsession.presentation.compilations
```
[the remark this answers is not in the document]
Comment (DimaK21) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsAdapter.kt]: сделал

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsViewHolder.kt [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+import androidx.recyclerview.widget.RecyclerView
+import com.davay.android.databinding.CompilationsItemBinding
+import com.davay.android.feature.createsession.domain.model.Compilation
+
+class CompilationsViewHolder(
+    private val binding: CompilationsItemBinding,
+    private val clickListener: CompilationsAdapter.ItemClickListener,
+) : RecyclerView.ViewHolder(binding.root) {
+    fun bind(compilation: Compilation) {
```
[the remark this answers is not in the document]
Comment (DimaK21) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsViewHolder.kt]: сделал

File: uikit/src/main/res/values/dimens.xml (hunk 3 above)
Comment (DimaK21) [reply uikit/src/main/res/values/dimens.xml]: переименовал

File: app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt (hunk 4 above)
Comment (DimaK21) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt:19]: надо проверять
я когда в других проектах подключал лик кэнэри, она на такие вещи ругалась

Comment (DimaK21) [reply app/src/main/java/com/davay/android/feature/createsession/presentation/compilations/CompilationsFragment.kt]: убрал

**Review by n-shvedov27 — APPROVED** [review]
